Tests fail with Python 3.10

Hello,

We've been trying to run django-auth-ldap tests on Fedora with Python 3.10 but 2 tests started to fail (3.9 is fine).

```
======================================================================
FAIL: test_deprecated_cache_groups (tests.tests.LDAPTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/django-auth-ldap-2.4.0/tests/tests.py", line 1578, in test_deprecated_cache_groups
    self.assertEqual(len(w), 1)
AssertionError: 5 != 1
======================================================================
FAIL: test_deprecated_callable_server_uri (tests.tests.LDAPTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/django-auth-ldap-2.4.0/tests/tests.py", line 195, in test_deprecated_callable_server_uri
    self.assertEqual(len(w), 1)
AssertionError: 5 != 1
----------------------------------------------------------------------
```

Can you take a look, please?

--- a/CHANGES
+++ b/CHANGES
@@ -1,6 +1,8 @@
 UNRELEASED
 ----------
 
+- Dropped deprecated setting ``AUTH_LDAP_CACHE_GROUPS``.
+- Callables passed to ``AUTH_LDAP_SERVER_URI`` must now take a ``request`` positional argument.
 - Dropped support for Django 3.0.
 
 2.4.0 - 2021-04-06

--- a/django_auth_ldap/backend.py
+++ b/django_auth_ldap/backend.py
@@ -59,7 +59,6 @@ from django.contrib.auth import get_user_model
 from django.contrib.auth.models import Group, Permission
 from django.core.cache import cache
 from django.core.exceptions import ImproperlyConfigured, ObjectDoesNotExist
-from django.utils.inspect import func_supports_parameter
 
 from django_auth_ldap.config import (
     ConfigurationWarning,
@@ -844,17 +843,7 @@ class _LDAPUser:
         if self._connection is None:
             uri = self.settings.SERVER_URI
             if callable(uri):
-                if func_supports_parameter(uri, "request"):
-                    uri = uri(self._request)
-                else:
-                    warnings.warn(
-                        "Update AUTH_LDAP_SERVER_URI callable %s.%s to accept "
-                        "a positional `request` argument. Support for callables "
-                        "accepting no arguments will be removed in a future "
-                        "version." % (uri.__module__, uri.__name__),
-                        DeprecationWarning,
-                    )
-                    uri = uri()
+                uri = uri(self._request)
 
             self._connection = self.backend.ldap.initialize(uri, bytes_mode=False)
 
@@ -1039,23 +1028,6 @@ class LDAPSettings:
         for name, default in defaults.items():
             value = getattr(django.conf.settings, prefix + name, default)
             setattr(self, name, value)
-
-        # Compatibility with old caching settings.
-        if getattr(
-            django.conf.settings,
-            self._name("CACHE_GROUPS"),
-            defaults.get("CACHE_GROUPS"),
-        ):
-            warnings.warn(
-                "Found deprecated setting AUTH_LDAP_CACHE_GROUP. Use "
-                "AUTH_LDAP_CACHE_TIMEOUT instead.",
-                DeprecationWarning,
-            )
-            self.CACHE_TIMEOUT = getattr(
-                django.conf.settings,
-                self._name("GROUP_CACHE_TIMEOUT"),
-                defaults.get("GROUP_CACHE_TIMEOUT", 3600),
-            )
 
     def _name(self, suffix):
         return self._prefix + suffix

--- a/docs/reference.rst
+++ b/docs/reference.rst
@@ -83,14 +83,6 @@ The value determines the amount of time, in seconds, a user's group memberships
 and distinguished name are cached. The value ``0``, the default, disables
 caching entirely.
 
-.. versionchanged:: 1.6.0
-
-    Previously caching was controlled by the settings `AUTH_LDAP_CACHE_GROUPS`
-    and `AUTH_LDAP_GROUP_CACHE_TIMEOUT`. If `AUTH_LDAP_CACHE_GROUPS` is set,
-    the `AUTH_LDAP_CACHE_TIMEOUT` value is derievd from these deprecated
-    settings.
-
-
 .. setting:: AUTH_LDAP_CONNECTION_OPTIONS
 
 AUTH_LDAP_CONNECTION_OPTIONS

--- a/tests/tests.py
+++ b/tests/tests.py
@@ -27,7 +27,6 @@ import functools
 import logging
 import os
 import pickle
-import warnings
 from copy import deepcopy
 from unittest import mock
 
@@ -177,29 +176,6 @@ class LDAPTest(TestCase):
         self.assertEqual(user.username, "alice")
         self.assertEqual(User.objects.count(), user_count + 1)
         cb_mock.assert_called_with(request)
-
-    def test_deprecated_callable_server_uri(self):
-        self._init_settings(
-            SERVER_URI=lambda: self.server.ldap_uri,
-            USER_DN_TEMPLATE="uid=%(user)s,ou=people,o=test",
-        )
-        user_count = User.objects.count()
-
-        with warnings.catch_warnings(record=True) as w:
-            warnings.simplefilter("always")
-            user = authenticate(username="alice", password="password")
-
-        self.assertIs(user.has_usable_password(), False)
-        self.assertEqual(user.username, "alice")
-        self.assertEqual(User.objects.count(), user_count + 1)
-        self.assertEqual(len(w), 1)
-        self.assertEqual(w[0].category, DeprecationWarning)
-        self.assertEqual(
-            str(w[0].message),
-            "Update AUTH_LDAP_SERVER_URI callable tests.tests.<lambda> to "
-            "accept a positional `request` argument. Support for callables "
-            "accepting no arguments will be removed in a future version.",
-        )
 
     def test_simple_bind(self):
         self._init_settings(USER_DN_TEMPLATE="uid=%(user)s,ou=people,o=test")
@@ -1559,29 +1535,6 @@ class LDAPTest(TestCase):
             self.assertIsNotNone(user)
         # Should have executed only once.
         self.assertEqual(mock.call_count, 1)
-        # DN is cached.
-        self.assertEqual(
-            cache.get("django_auth_ldap.user_dn.alice"), "uid=alice,ou=people,o=test"
-        )
-
-    def test_deprecated_cache_groups(self):
-        self._init_settings(
-            USER_SEARCH=LDAPSearch(
-                "ou=people,o=test", ldap.SCOPE_SUBTREE, "(uid=%(user)s)"
-            ),
-            CACHE_GROUPS=True,
-        )
-        with warnings.catch_warnings(record=True) as w:
-            warnings.simplefilter("always")
-            user = authenticate(username="alice", password="password")
-        self.assertIsNotNone(user)
-        self.assertEqual(len(w), 1)
-        self.assertEqual(w[0].category, DeprecationWarning)
-        self.assertEqual(
-            str(w[0].message),
-            "Found deprecated setting AUTH_LDAP_CACHE_GROUP. Use "
-            "AUTH_LDAP_CACHE_TIMEOUT instead.",
-        )
         # DN is cached.
         self.assertEqual(
             cache.get("django_auth_ldap.user_dn.alice"), "uid=alice,ou=people,o=test"

